Yatima/Transpiler/LurkFunctions.lean

```diff
@@ -38,6 +32,15 @@ def append : Name × AST := (`append, ⟦
   ))
 ⟧)
 
+def str_append : Name × AST := (`str_append, ⟦
+  (lambda (xs ys)
+    (if (eq "" xs)
+      ys
+      (strcons
+        (car xs)
+        (str_append (cdr xs) ys))))
+⟧)
+
```

Comment:
```suggestion
def StringAppend : Name × AST := (``String.append, ⟦
  (lambda (xs ys)
    (if (eq "" xs)
      ys
      (strcons
        (car xs)
        (String.append (cdr xs) ys))))
⟧)
```
We should keep the convention that any replaced Lean function should be called by the same name. In 99% of cases `String.append` will be called through `HAppend`, e.g. `a ++ b`, but if someone actually writes `String.append a b`, we will accidently transpile the Lean one as well and pollute the functions. This is why all the builtins I added have the names they do.